pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -732,7 +753,32 @@ func (dr DashboardServiceImpl) CountInFolders(ctx context.Context, orgID int64,
 }
 
 func (dr *DashboardServiceImpl) DeleteInFolders(ctx context.Context, orgID int64, folderUIDs []string, u identity.Requester) error {
-	return dr.dashboardStore.DeleteDashboardsInFolders(ctx, &dashboards.DeleteDashboardsInFolderRequest{FolderUIDs: folderUIDs, OrgID: orgID})
+	for _, folderUID := range folderUIDs {
+		// only hard delete the folder representation in the dashboard store
+		// nolint:staticcheck
+		if err := dr.dashboardStore.SoftDeleteDashboardsInFolder(ctx, orgID, folderUID); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
 func (dr *DashboardServiceImpl) Kind() string { return entity.StandardKindDashboard }
+
+func (dr *DashboardServiceImpl) CleanUpDeletedDashboards(ctx context.Context) (int64, error) {
+	var deletedDashboardsCount int64
```

Comment:
I thought this could help to clean up Soft deleted dashboards before feature flag disable.
Like: 
* Enabling the feature on an instance
* Deleting a couple of dashboards
* Disabling the feature

In this particular case, those dashboards will live forever on the DB. If we leave this background service turned on it could remove those from the DB.
But if you see a problem of going this I can remove it from the entry point (not running the background service when the feature is disabled)